lib/model/unittest/CMetricPopulationModelTest.cc

```diff
@@ -1226,54 +1223,100 @@ BOOST_FIXTURE_TEST_CASE(testIgnoreSamplingGivenDetectionRules, CTestFixture) {
+    // different weightings applied to the samples for attribute c4
+    uint64_t withSkipChecksum = modelWithSkip->checksum();
+    uint64_t noSkipChecksum = modelNoSkip->checksum();
+    BOOST_TEST_REQUIRE(withSkipChecksum != noSkipChecksum);
+
+    // expect models for attributes c0 - c4
+    for (std::size_t i = 0; i < 5; ++i) {
+        mathsModelWithSkipView = modelWithSkipView->model(
+            model_t::E_PopulationMeanByPersonAndAttribute, i);
+        BOOST_TEST_REQUIRE(mathsModelWithSkipView != nullptr);
+
+        mathsModelNoSkipView =
+            modelNoSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, i);
+        BOOST_TEST_REQUIRE(mathsModelNoSkipView != nullptr);
+    }
 
-    CAnomalyDetectorModel::TModelDetailsViewUPtr modelWithSkipView =
-        modelWithSkip->details();
-    CAnomalyDetectorModel::TModelDetailsViewUPtr modelNoSkipView = modelNoSkip->details();
+    // The underlying models for attributes c0 - c3 are the same
+    for (std::size_t i = 0; i < 4; ++i) {
+        mathsModelWithSkipView = modelWithSkipView->model(
+            model_t::E_PopulationMeanByPersonAndAttribute, i);
+        withSkipChecksum = mathsModelWithSkipView->checksum();
+        mathsModelNoSkipView =
+            modelNoSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, i);
+        noSkipChecksum = mathsModelNoSkipView->checksum();
+        BOOST_REQUIRE_EQUAL(withSkipChecksum, noSkipChecksum);
+    }
+
+    // While the underlying models for attribute c4 differ due to the different weighting applied to the samples
+    mathsModelWithSkipView =
+        modelWithSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, 4);
+    BOOST_TEST_REQUIRE(mathsModelWithSkipView != nullptr);
+    withSkipChecksum = mathsModelWithSkipView->checksum();
+
+    mathsModelNoSkipView =
+        modelNoSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, 4);
+    BOOST_TEST_REQUIRE(mathsModelNoSkipView != nullptr);
+    noSkipChecksum = mathsModelNoSkipView->checksum();
 
-    // but the underlying models for people p1 and p2 are the same
-    uint64_t withSkipChecksum = modelWithSkipView
-                                    ->model(model_t::E_PopulationMeanByPersonAndAttribute, 0)
-                                    ->checksum();
-    uint64_t noSkipChecksum =
-        modelNoSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, 0)->checksum();
-    BOOST_REQUIRE_EQUAL(withSkipChecksum, noSkipChecksum);
-
-    withSkipChecksum = modelWithSkipView
-                           ->model(model_t::E_PopulationMeanByPersonAndAttribute, 1)
-                           ->checksum();
-    noSkipChecksum =
-        modelNoSkipView->model(model_t::E_PopulationMeanByPersonAndAttribute, 1)->checksum();
-    BOOST_REQUIRE_EQUAL(withSkipChecksum, noSkipChecksum);
+    BOOST_TEST_REQUIRE(withSkipChecksum != noSkipChecksum);
 
     // TODO These checks fail see elastic/ml-cpp/issues/2043
```

Comment:
Again, it seems like the issue doesn't match what's commented out, because the issue is for checksums while the commented out code is for last value times.  Or am I missing something?